src/main/scala/com/fulcrumgenomics/bam/api/SamSource.scala

```diff
@@ -91,9 +91,16 @@ class SamSource private(private val reader: SamReader) extends View[SamRecord] w
   /** Returns an iterator over all the records in the source. */
   override def iterator: SamIterator = new SamIterator(reader.getFileHeader, reader.iterator())
 
+  private def newQueryInterval(region: Locatable): QueryInterval = {
+    val contig = dict(region.getContig)
```

Comment:
```suggestion
    val contig = dict.get(region.getContig).getOrElse(
      throw new NoSuchElementException("Contig '${region.getContig}' not in SAM/BAM header.")
```

Or pick your favorite exception to throw .. just makes the error nicer if you query a contig that's not in the BAM header.